subprojects/docs/src/docs/userguide/workingWithFiles.adoc

```diff
@@ -64,58 +347,101 @@ You can also pass the `files()` method a closure or a `Callable` instance. This
 </sample>
 ++++
 
-Some other types of things you can pass to `files()`:
+The key to lazy creation is passing a closure to the `files()` method. Your closure simply needs to return a value of a type accepted by `files()`, such as `List<File>`, `String`, `FileCollection`, etc.
 
-`FileCollection`::
-These are flattened and the contents included in the file collection.
-`Task`::
-The output files of the task are included in the file collection.
-`TaskOutputs`::
-The output files of the TaskOutputs are included in the file collection.
+_Iterating over a file collection_ can be done through the `each()` method on the collection or using the collection in a `for` loop. In both approaches, the file collection is treated as a set of `File` instances, i.e. your iteration variable will be of type `File`.
 
+The following example demonstrates such iteration as well as how you can convert file collections to other types using the `as` operator or supported properties:
 
-It is important to note that the content of a file collection is evaluated lazily, when it is needed. This means you can, for example, create a `FileCollection` that represents files which will be created in the future by, say, some task.
+++++
+<sample id="fileCollections" dir="userguide/files/fileCollections" title="Using a file collection">
+    <sourcefile file="build.gradle" snippet="usage"/>
+    <test args="-q usage"/>
+</sample>
+++++
+
+You can also see at the end of the example _how to combine file collections_ using the `+` and `-` operators to merge and subtract them. An important feature of the resulting file collections is that they are _live_. In other words, when you combine file collections in this way, the result always reflects what's in the original file collections, even if they change during the build.
```

Comment:
Is there any more detailed information on what you're talking about? Modifying a live collection in what way? By adding/removing elements or modifying the elements themselves?

 - [ ] Warn against modifying live collections

Reply:
@eriwen Do you have any more information on this? I don't know what is specifically inefficient.

Reply:
@pledbrook Forgive me, I thought a [build file like this](https://gist.github.com/eriwen/5e41be63170611d40358f7e80847a381) would cause an infinite loop, but it does not. No action needed from you here.